components/scream/src/share/field/field.hpp

```diff
@@ -464,6 +468,62 @@ sync_to_dev () const {
   Kokkos::deep_copy(m_view_d,*m_view_h);
 }
 
+template<typename RealType>
+template<HostOrDevice HD>
+void Field<RealType>::
+set_value (const field_type& field_src) {
+  const auto& layout     = get_header().get_identifier().get_layout();
+  const auto& layout_src = field_src.get_header().get_identifier().get_layout();
+  EKAT_REQUIRE_MSG(layout==layout_src,
+       "ERROR: Unable to copy field " + field_src.get_header().get_identifier().name() + 
+          " to field " + get_header().get_identifier().name() + " during initialization.  Layouts don't match.");
+  const auto  rank = layout.rank();
+  // Note: we can't just do a deep copy on get_view<HD>(), since this
+  //       field might be a subfield of another. So check the header
+  //       first, to see if we have a parent. If not, deep copy is fine.
+  //       If we do, we need to get the reshaped view first.
+  const auto parent = get_header().get_parent();
+  if (parent.lock()==nullptr) {
+    // No parent. Deep copying to get_view<HD>() is safe.
+    Kokkos::deep_copy (get_view<HD>(),field_src.get_view<HD>());
```

Comment:
What if `field_src` has a parent? This code would crash.

I think we might as well get rid of the parent check, and simply call `get_reshaped_view` on both `*this` and `field_src`.